tensorflow/lite/micro/kernels/svdf_test.cc

```diff
@@ -657,6 +659,9 @@ inline void TestIntegerSVDF(
 
 TF_LITE_MICRO_TESTS_BEGIN
 
+#if !defined(XTENSA)  // TODO(b/170332589): xtensa kernels are less general than
+                      // reference kernels and we ifdef out test cases that are
+                      // currently known to fail.
```

Comment:
I see one potential difficulty with this approach.  The float variants are not supported on HiFi mini. But they can be supported on other cores like HiFi 4 or HiFi 5, for these we would like these tests to be enabled.
Instead of a platform specific check like #if !defined(XTENSA), can we add a 'test specific' check like #if !defined(DISABLE_SVDF_FLOAT_TESTS) ? We could then selectively disable the tests based on TARGET_ARCH in xtensa_makefile.inc.

Reply:
That's a good point. We can incrementally make this check more specific (i.e. `#if !defined(XTENSA_HIFIMINI)` as needed).

What I'd like to have is a consistent way to determine which cases are supported for a particular target while reading the test code. With `DISABLE_SVDF_FLOAT_TESTS`, someone reading the code would have to then look through the makefiles to see which targets define this. And there would be a separate define for `DISABLE_CONV_FLOAT_TESTS` etc.

And the medium-term goal is to hopefully have all test cases pass on all targets with hifimini being the exception rather than the norm.

Does that seem ok to you?

Reply:
Yes, sounds good.
`#if !defined(XTENSA_HIFIMINI)` will take care of unsupported hifimini testcases. For other cores we will have most of the tests passing, either as optimized implementation or the reference implementation.